eth_utils/abi.py

```diff
@@ -42,27 +83,682 @@ def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+        >>> abi = {
+        ...   'constant': False,
+        ...   'inputs': [
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'name': 'f',
+        ...   'outputs': [],
+        ...   'payable': False,
+        ...   'stateMutability': 'nonpayable',
+        ...   'type': 'function'
+        ... }
+        >>> abi_to_signature(abi)
+        'f(uint256)'
+    """
+    if abi["type"] == "fallback" or abi["type"] == "receive":
+        name = str(abi["type"])
+        abi_inputs = None
+    elif abi["type"] == "constructor":
+        name = abi["type"]
+        abi_inputs = abi["inputs"]
+    elif abi["type"] == "event":
+        name = abi["name"]
+        abi_inputs = abi["inputs"]
+    else:
+        name = str(abi.get("name", abi["type"]))
+        abi_inputs = abi.get("inputs")
+
+    return f"{name}({_abi_inputs_types(abi_inputs)})"
+
+
+def filter_by_type(type: str, contract_abi: ABI) -> List[ABIElement]:
+    """
+    Return a list of each ``ABIElement`` that is of type ``_type``.
+
+    :param type: Type of ABI element to filter by.
+    :type type: `str`
+    :param contract_abi: Contract ABI.
+    :type contract_abi: `ABI`
+    :return: List of ABI elements of the specified type.
+    :rtype: `List[ABIElement]`
+
+    .. doctest::
+
+        >>> from eth_utils import filter_by_type
+        >>> abi = [
+        ...   {"type": "function", "name": "myFunction", "inputs": [], "outputs": []},
+        ...   {"type": "function", "name": "myFunction2", "inputs": [], "outputs": []},
+        ...   {"type": "event", "name": "MyEvent", "inputs": []}
+        ... ]
+        >>> filter_by_type("function", abi)
+        [{'type': 'function', 'name': 'myFunction', 'inputs': [], 'outputs': []}, \
+{'type': 'function', 'name': 'myFunction2', 'inputs': [], 'outputs': []}]
+    """
+    return [abi for abi in contract_abi if abi["type"] == type]
+
+
+def get_all_function_abis(abi: ABI) -> Sequence[ABIFunction]:
+    """
+    Return interfaces for each function in the contract ABI.
+
+    :param abi: Contract ABI.
+    :type abi: `ABI`
+    :return: List of ABIs for each function interface.
+    :rtype: `list[ABIFunction]`
+
+    .. doctest::
+
+        >>> from eth_utils import get_all_function_abis
+        >>> abi = [
+        ...   {"type": "function", "name": "myFunction", "inputs": [], "outputs": []},
+        ...   {"type": "function", "name": "myFunction2", "inputs": [], "outputs": []},
+        ...   {"type": "event", "name": "MyEvent", "inputs": []}
+        ... ]
+        >>> get_all_function_abis(abi)
+        [{'type': 'function', 'name': 'myFunction', 'inputs': [], 'outputs': []}, \
+{'type': 'function', 'name': 'myFunction2', 'inputs': [], 'outputs': []}]
+    """
+    return [
+        cast(ABIFunction, function_abi)
+        for function_abi in filter_by_type("function", abi)
+    ]
+
+
+def get_all_event_abis(abi: ABI) -> Sequence[ABIEvent]:
```

Comment:
nit for clarity: If this expects the whole contract ABI, as the `:param` below says:

```suggestion
def get_all_event_abis(contract_abi: ABI) -> Sequence[ABIEvent]:
```